### PR TITLE
feat: add publish dockerfile

### PR DESCRIPTION
**Pull request by ntindle**
publish the dockerfile to aaa-langs
move the Dockerfile to the root of the repo

### CODE REVIEW

**Comment by ntindle** [comment]
use the below command to test after logging in with [these instructions](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry)

```bash
docker run -v $(pwd):/tmp -it ghcr.io/algorithm-archivists/aaa-langs:feature-publish-container bash
```

**Comment by ntindle** [comment]
@ShadowMitia can you take a look?

**Comment by ntindle** [comment]
Can @algorithm-archivists/development give this an approval?

File: .github/workflows/publish_container.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Publish Docker
+on: [push]
```
Comment (ShadowMitia) [inline .github/workflows/publish_container.yml:2]: I've seen other examples of github actions where they restrict branches this is run on. I'm thinking maybe restrict to code pushed to master?
```suggestion
on:
  push:
    branches:
      - master
```

**Review by ShadowMitia — APPROVED** [review]
Looks good to me.

I just have a suggestion for restricting building the image on master branch.

While we're at it though, maybe now would be a good time to change to `main` before we do any changes that depend on the branch name.

File: .github/workflows/publish_container.yml (hunk 1 above)
Comment (ntindle) [reply .github/workflows/publish_container.yml:2]: we can do that, I had to leave it out to test it

**Review by ShadowMitia — APPROVED** [review]
Looks good to me. I'm still suggesting we do the full change to `main` before, but otherwise I think we're good.

**Review by leios — APPROVED** [review]
I also think it's fine and am happy to approve it

**Comment by ntindle** [comment]
`docker run -v $(pwd):/tmp -it ghcr.io/algorithm-archivists/aaa-langs:latest bash`

**Pull request merged**